packages/build-utils/src/lambda.ts

```diff
@@ -305,6 +316,13 @@ export class Lambda {
       }
     }
 
+    if (supportsCancellation !== undefined) {
+      assert(
+        typeof supportsCancellation === 'boolean',
+        '"supportsCancellation" is not a boolean'
+      );
+    }
+
```

Comment:
Might be worth double checking if this is not already covered in our schema validation

https://github.com/vercel/vercel/blob/f5d6674b41467653c820c90ae769643e141014fb/packages/cli/src/util/validate-config.ts#L158-L178